components/header.js

```diff
@@ -1,67 +1,218 @@
-      align-items: center;
-      column-gap: 40px;
-    }
-
-    a {
-      display: flex;
-      gap: 10px;
-      align-items: center;
-      text-decoration: none;
-    }
-
-    a img {
-      width: 45px;
-      height: 22.5px;
-    }
-
-    a span {
-      font-size: 20px;
-      font-weight: 500;
-      color: black;
-    }
-  </style>
-  <header>
-      <a href="/">
-        <img src="images/logo.png" alt="logo" />
-        <span>달레 스터디</span>
-      </a>
-    
-      <div class="buttons-container">
-        <ds-button-link href="#join-instruction-container" size="small" variant="ghost">참여방법 안내</ds-button-link>
-        <ds-button-link href="https://discord.gg/43UkheRV" size="small" variant="primary">디스코드 참여하기</ds-button-link>
-      </div>
-  </header>
-`;
-
-class HeaderComponent extends HTMLElement {
+import { css, html } from "../html-css-utils.js";
+
+class Header extends HTMLElement {
   constructor() {
     super();
+    this.render();
+    this.setupEventListeners();
+  }
+
+  setupEventListeners() {
+    const menuButton = this.shadowRoot.querySelector("button");
+    menuButton.addEventListener("click", this.toggleMenu.bind(this));
+  }
+
+  toggleMenu() {
+    const buttonLinks = this.shadowRoot.querySelector(".buttons-container");
+    const header = this.shadowRoot.querySelector("header");
+    const menuIcon = this.shadowRoot.querySelector("#menuIcon");
+    const closeIcon = this.shadowRoot.querySelector("#closeIcon");
+
+    menuIcon.classList.toggle("hide");
+    closeIcon.classList.toggle("hide");
+
+    buttonLinks.classList.toggle("open");
+    header.classList.toggle("vertical");
+    this.classList.toggle("menu-open");
+  }
+
+  render() {
     this.attachShadow({ mode: "open" });
-    this.shadowRoot.appendChild(template.content.cloneNode(true));
+    this.shadowRoot.innerHTML = this.createCss() + this.createHtml();
+  }
+
+  createCss() {
+    return css`
+      :host {
+        position: fixed;
+        top: 0;
+        left: 0;
+        color: var(--text-900);
+        padding: 20px 0;
+        font-weight: var(--font-weight-medium);
+        width: 100%;
+        border-bottom: 1px solid var(--bg-300);
+        background-color: var(--bg-200);
+        @media only screen and (min-width: 1024px) {          
+          z-index: 1000;
+        }
+      }
+
+      :host(.menu-open) {
+        background-color: var(--bg-100);
+      }
+
+      *:focus {
+        border: 1px solid black;
+      }
+
+      header {
+        width: 80%;
+        margin: 0 auto;
+        justify-content: space-between;
+        display: flex;
+        align-items: center;
+
+        @media only screen and (min-width: 768px) {
+          flex-direction: row !important;
+        }
+      }
+
+      header.vertical {
+        flex-direction: column;
+      }
+
+      .buttons-container {
+        display: none;
+        align-items: center;
+        column-gap: 40px;
+        @media only screen and (min-width: 768px) {
```

Comment:
가독성과 일관성을 위해 한 줄 개행하면 어떨까요?

```suggestion
        column-gap: 40px;
        
        @media only screen and (min-width: 768px) {
```

Reply:
이 부분까진 prettier가 커버해주지 않군요. 기행하고 다시 force push 하겠습니다!